lucene/core/src/java/org/apache/lucene/util/bkd/BKDReader.java

```diff
@@ -498,6 +500,7 @@ public IntersectState(IndexInput in, int numDims,
       this.scratchMinIndexPackedValue = new byte[packedIndexBytesLength];
       this.scratchMaxIndexPackedValue = new byte[packedIndexBytesLength];
       this.index = indexVisitor;
+      this.scratchLongs = new long[maxPointsInLeafNode / 2];
```

Comment:
does it need to be 
```suggestion
      this.scratchLongs = new long[(maxPointsInLeafNode + 1) / 2];
```
so that it keeps working for even numbers of points per node?

Reply:
Not really. We read as many longs as we can so if the count is un-even for 32bits integer, the last four bytes are read from the index. See Line 120 of DocIdsWriter.